lock-api/src/test/java/com/palantir/lock/client/LeasingTimelockClientTest.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock.client;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import java.time.Duration;
+import java.util.Set;
+import java.util.UUID;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import com.google.common.collect.ImmutableSet;
+import com.palantir.common.time.NanoTime;
+import com.palantir.lock.v2.ImmutableIdentifiedTimeLockRequest;
+import com.palantir.lock.v2.LeaderTime;
+import com.palantir.lock.v2.LeadershipId;
+import com.palantir.lock.v2.Lease;
+import com.palantir.lock.v2.LockImmutableTimestampResponse;
+import com.palantir.lock.v2.LockRequest;
+import com.palantir.lock.v2.LockResponse;
+import com.palantir.lock.v2.LockResponseV2;
+import com.palantir.lock.v2.LockToken;
+import com.palantir.lock.v2.RefreshLockResponseV2;
+import com.palantir.lock.v2.StartAtlasDbTransactionResponseV3;
+import com.palantir.lock.v2.StartIdentifiedAtlasDbTransactionRequest;
+import com.palantir.lock.v2.StartIdentifiedAtlasDbTransactionResponse;
+import com.palantir.lock.v2.TimelockRpcClient;
+import com.palantir.lock.v2.TimelockService;
+import com.palantir.lock.v2.TimestampAndPartition;
+
+@RunWith(MockitoJUnitRunner.class)
+public class LeasingTimelockClientTest {
+    @Mock private TimelockRpcClient timelockRpcClient;
+    @Mock private LockRequest lockRequest;
+    @Mock private TimestampAndPartition timestampAndPartition;
+
+    private TimelockService timelockService;
+    private static final LeadershipId LEADER_ID = LeadershipId.random();
+    private static final UUID SERVICE_ID = UUID.randomUUID();
+    private StartIdentifiedAtlasDbTransactionRequest startTxnRequest =
+            StartIdentifiedAtlasDbTransactionRequest.createForRequestor(SERVICE_ID);
+
+    private static final LockToken LOCK_TOKEN = LockToken.of(UUID.randomUUID());
+    private static final LockResponse LOCK_RESPONSE = LockResponse.successful(LOCK_TOKEN);
+
+    @Before
+    public void before() {
+        when(timelockRpcClient.getLeaderTime()).thenReturn(LeaderTime.of(LeadershipId.random(), NanoTime.now()));
```

Comment:
red flag: the fact you're not ever changing the time implies that this testing is insufficient. The purpose of writing unit tests like this is to codify the assumptions you're making about how this code should work; the time being static here is a huge red flag to me because it means that all these tests are time invariant (which means that there's no leasing being tested).